.github/workflows/tests.yaml

```diff
@@ -227,12 +248,15 @@ jobs:
 
       # We need to replace the `.` with a `_` for the build cache.
       - name: Setup LLVM and Clang (macOS)
-        if: steps.filter.outputs.has_code == 'true' && matrix.os == 'macos-12'
+        if:
+          steps.filter.outputs.has_code == 'true' && contains(matrix.os,
+          'macos')
         run: |
           echo "os_for_cache=macos-12" >> $GITHUB_ENV
       - name: Setup LLVM and Clang (Ubuntu)
         if:
-          steps.filter.outputs.has_code == 'true' && matrix.os == 'ubuntu-22.04'
+          steps.filter.outputs.has_code == 'true' && contains(matrix.os,
+          'ubuntu')
```

Comment:
Similar to the macos-12 comment, are you sure all ubuntu builds are sufficiently hermetic that they will yield the same outputs regardless of version? Note, I think we've seen some corrupt builds from caching of non-hermetic build outputs before -- IIRC that's https://github.com/carbon-language/carbon-lang/pull/2198 and its note about a slight difference in include paths.

Reply:
Note, maybe if build hermeticism issue is solved, we could just remove the os suffix now?

Reply:
No, they're not. Adding ARM works "fine", but by accident -- the *CPU* part of this is part of the action hash used to query the cache. But the OS versions are not. Nor are the distinctions between Ubuntu and some other Linux distro if we ever care.

But there *is* a great way to solve the hermeticity issues here more simply than partitioning our buckets! We can use the `--action_env` flag to thread a key for the remote cache in from the workflow file, and set that in a way that allows us to use a single bucket.

I've done this to make it more obvious that the two macOS runs are correct, and left a TODO to completely remove this -- that seems like it's worth keeping as a separate change as we'll need to fix up the name of the buckets, the keys, and then do a round of fresh builds to re-populate them.

Reply:
I've created a carbon-builds-github-v1 is you want to switch to just a single bucket here. (just dropping the os suffix)

> that seems like it's worth keeping as a separate change as we'll need to fix up the name of the buckets, the keys, and then do a round of fresh builds to re-populate them.

I think all you'd need to do is change the names of the bucket in this PR. Fresh builds are also necessary with either the macos-14 or the --action_env addition (both require cache population), you'd just be putting builds into a different bucket.

Keys are unaffected since it's still the same account.

Reply:
Ah cool, nice that it's that easy. I've done this and tested that it all works, and we get caching afterward.